go/worker/compute/executor/committee/orderedmap/ordered_map.go

```diff
@@ -0,0 +1,194 @@
+// Package orderedmap implements a queue backed by an ordered map.
+package orderedmap
+
+import (
+	"container/list"
+	"fmt"
+	"sync"
+
+	"github.com/hashicorp/go-multierror"
+
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/hash"
+	"github.com/oasisprotocol/oasis-core/go/runtime/scheduling/simple/txpool/api"
```

Comment:
So this is tied to the simple scheduler (due to the error type), maybe it should go into a package under `simple`? It seems similar to the txpool API but not quite the same.